src/components/App.tsx

```diff
@@ -1,36 +1,25 @@
+import { Entity } from '@backstage/catalog-model';
 import React from 'react';
-import { Route, MemoryRouter, Routes } from 'react-router';
-import { BuildsPage, Builds } from '../pages/BuildsPage';
-import { AppStateProvider } from '../state';
-import { Settings } from './Settings';
+import { Builds } from '../components/BuildsPage';
+import { ContextProvider } from './ContextProvider';
+import { LastBuildCard } from './LastBuildCard';
 
-export const App = () => {
+type Props = {
+  entity: Entity;
+};
+
+export const App: React.FC<Props> = ({ entity }) => {
   return (
-    <AppStateProvider>
-      <>
-        <Routes>
-          <Route path="/" element={<BuildsPage />} />
-        </Routes>
-        <Settings />
-      </>
-    </AppStateProvider>
+    <ContextProvider entity={entity}>
+      <Builds />
+    </ContextProvider>
   );
 };
 
-// TODO: allow pass in settings as props
-// When some shared settings workflow
-// will be established
-export const TravisCIWidget = () => {
+export const TravisCIWidget: React.FC<Props> = ({ entity }) => {
```

Comment:
Can we extract this to its own file? And give the component a more specific name ? Like RecentTravisCIBuildsWidget or something to that effect if you think that's too long.